src/app/parcels/batch/displayComponents/PersonEditCellInput.tsx

```diff
@@ -0,0 +1,127 @@
+import { PERSON_WIDTH } from "@/app/parcels/batch/columnWidths";
+import { BatchActionType } from "@/app/parcels/batch/batchTypes";
+import { PersonField } from "@/app/parcels/batch/displayComponents/PersonEditCell";
+import { Person } from "@/components/Form/formFunctions";
+import dayjs from "dayjs";
+import { MINIMUM_NUMBER_OF_ADULTS, MAXIMUM_NUMBER_OF_ADULTS } from "@/app/clients/form/bounds";
+import PersonInput from "@/app/parcels/batch/displayComponents/PersonInput";
+import { ChangeEvent, useEffect, useState } from "react";
+import {
+    BottomDiv,
+    EditCellInputLabel,
+    EditCellTextField,
+    PersonInputDiv,
+} from "@/app/parcels/batch/displayComponents/EditCellStyledComponents";
+
+interface PersonEditCellInputProps {
+    id: number;
+    dispatchBatchTableAction: React.Dispatch<BatchActionType>;
+    personField: PersonField;
+    personArray: Person[];
+}
+
+const PersonEditCellInput: React.FC<PersonEditCellInputProps> = ({
+    id,
+    dispatchBatchTableAction,
+    personField,
+    personArray,
+}) => {
+    const [people, setPeople] = useState<Person[]>([]);
+    useEffect(() => {
+        setPeople(personArray);
+    }, [personArray]);
+    const MIN_NUMBER_PEOPLE =
+        personField === "childrenInfo" || id === 0 ? 0 : MINIMUM_NUMBER_OF_ADULTS;
+    const MAX_NUMBER_PEOPLE = MAXIMUM_NUMBER_OF_ADULTS;
+
+    const dispatchPerson = (newPoeple: Person[]): void => {
+        personField === "adultInfo"
+            ? dispatchBatchTableAction({
+                  type: "update_cell",
+                  updateCellPayload: {
+                      rowId: id,
+                      newValueAndFieldName: {
+                          type: "client",
+                          fieldName: personField,
+                          newValue: {
+                              adults: newPoeple,
+                              numberOfAdults: newPoeple.length,
+                          },
+                      },
+                  },
+              })
+            : dispatchBatchTableAction({
+                  type: "update_cell",
+                  updateCellPayload: {
+                      rowId: id,
+                      newValueAndFieldName: {
+                          type: "client",
+                          fieldName: personField,
+                          newValue: {
+                              children: newPoeple,
+                              numberOfChildren: newPoeple.length,
+                          },
+                      },
+                  },
+              });
+    };
```

Comment:
Firstly, typo in `newPeople`. Secondly, I think we can simplify this by defining `newValue` conditionally then having oly one dispatch call:
```
    const dispatchPerson = (newPoeple: Person[]): void => {
        const newValue =  personField === "adultInfo"
            ? {
                 adults: newPoeple,
                 numberOfAdults: newPoeple.length,
             }
            : {
                 children: newPoeple,
                 numberOfChildren: newPoeple.length,
             },
        dispatchBatchTableAction({
                  type: "update_cell",
                  updateCellPayload: {
                      rowId: id,
                      newValueAndFieldName: {
                          type: "client",
                          fieldName: personField,
                          newValue
                      },
                  },
              })
    };
```